src/main/java/org/apache/sysml/hops/UnaryOp.java

```diff
@@ -563,6 +569,10 @@ protected double computeIntermediateMemEstimate( long dim1, long dim2, long nnz
 			// getMemEstimate works for both cases of known dims and worst-case stats
 			ret = getInput().get(0).getMemEstimate() * 3; 
 		}
+
+		if (DMLScript.USE_ACCELERATOR) {
+			OptimizerUtils.estimateSize(dim1, dim2); // Intermediate memory required to convert sparse to dense
+		}
 		
 		return ret;
 	}
```

Comment:
This is ok for now but I'd like to get a reference for this. I could construct scenarios with small-ish output where partial aggregations are usually advantageous but this requires additional memory even for dense inputs. Anyway, if we do not have any details about these internals there is probably nothing we can do about it.